src/components/app/pageEvents/components/eventDialogContent.tsx

```diff
@@ -1,56 +1,152 @@
-
-const NEXT_PUBLIC_GOOGLE_CIVIC_API_KEY = requiredEnv(
-  process.env.NEXT_PUBLIC_GOOGLE_CIVIC_API_KEY,
-  'NEXT_PUBLIC_GOOGLE_CIVIC_API_KEY',
-)
+import { SWCEvent } from '@/utils/shared/getSWCEvents'
+import { USER_ACTION_TO_CAMPAIGN_NAME_DEFAULT_MAP } from '@/utils/shared/userActionCampaigns'
+import { triggerServerActionForForm } from '@/utils/web/formUtils'
+import { identifyUserOnClient } from '@/utils/web/identifyUser'
+import { toastGenericError } from '@/utils/web/toastUtils'
 
 export enum SectionNames {
   EVENT_INFO = 'Event Information',
+  PHONE_SECTION = 'Phone Section',
   NOTIFICATION_ACTIVATED = 'Event Notification Activated',
 }
 
 interface EventDialogContentProps {
-  event: SWCEvents[0]['data']
+  event: SWCEvent
 }
 
 export function EventDialogContent({ event }: EventDialogContentProps) {
+  usePreventOverscroll()
+  const { data: userFullProfileInfoResponse } = useApiResponseForUserFullProfileInfo()
+  const { user } = userFullProfileInfoResponse ?? { user: null }
+  const hasPhoneInformation = !!user?.phoneNumber
+  const [isCreatingRsvpEventAction, setIsCreatingRsvpEventAction] = useState(false)
+
   const sectionProps = useSections({
     sections: Object.values(SectionNames),
     initialSectionId: SectionNames.EVENT_INFO,
     analyticsName: 'Event Details Dialog',
   })
 
+  async function handleCreateRsvpAction({
+    shouldReceiveNotifications,
+  }: {
+    shouldReceiveNotifications: boolean
+  }) {
+    setIsCreatingRsvpEventAction(true)
+
+    const data: CreateActionRsvpEventInput = {
+      eventSlug: event.slug,
+      eventState: event.state,
+      shouldReceiveNotifications,
+    }
+
+    const result = await triggerServerActionForForm(
+      {
+        formName: 'RSVP Event',
+        onError: (_, error) => {
+          toast.error(error.message, {
+            duration: 5000,
+          })
+        },
+        analyticsProps: {
+          'Campaign Name': USER_ACTION_TO_CAMPAIGN_NAME_DEFAULT_MAP.RSVP_EVENT,
+          'User Action Type': UserActionType.RSVP_EVENT,
+          eventSlug: event.slug,
+          eventState: event.state,
+          shouldReceiveNotifications,
+        },
+        payload: data,
+      },
+      payload =>
+        actionCreateUserActionRsvpEvent(payload).then(actionResult => {
+          if (actionResult?.user) {
+            identifyUserOnClient(actionResult.user)
+          }
+          return actionResult
+        }),
+    )
+
+    if (result.status !== 'success') {
+      toastGenericError()
+    }
+    setIsCreatingRsvpEventAction(false)
+  }
+
+  async function handleGetUpdates() {
+    await handleCreateRsvpAction({
+      shouldReceiveNotifications: true,
+    })
+
+    toast.success('Thank you for your interest! We will send you a reminder closer to the event.')
+    sectionProps.goToSection(SectionNames.NOTIFICATION_ACTIVATED)
+  }
+
+  async function handleRSVPButtonClick() {
+    await handleCreateRsvpAction({
+      shouldReceiveNotifications: false,
+    })
```

Comment:
Here is what we talked about making it async. Just change to:

```suggestion
    void handleCreateRsvpAction({
      shouldReceiveNotifications: false,
    })
```

This will trigger the create action but will not block the code.

Reply:
Updated!